pallets/rmrk-core/src/functions.rs

```diff
@@ -232,11 +249,10 @@ where
 		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
 		Nfts::<T>::remove(collection_id, nft_id);
 
-		for _ in Resources::<T>::drain_prefix((collection_id, nft_id)) {}		
+		for _ in Resources::<T>::drain_prefix((collection_id, nft_id)) {}
 
-		let kids = Children::<T>::take((collection_id, nft_id));
-		for (child_collection_id, child_nft_id) in kids {
-			// Remove child from Children StorageMap
+		let c = Children::<T>::iter_prefix((collection_id, nft_id,)).count();
```

Comment:
Can we instead  `drain_prefix` and no longer need `remove_child`:
```
for ((child_collection_id, child_nft_id), _) in Children::<T>::drain_prefix((collection_id, nft_id)) {
    Self::nft_burn(child_collection_id, child_nft_id, max_recursions - 1)?;
}
```

Reply:
as @h4x3rotab mentioned in some other PR just `drain_prefix` could be enough in this case, so `Self::nft_burn` not needed?

Reply:
i switch to @HashWarlock's recommendation and it worked.  Will try dropping the Self::nft_burn next.  Would love to avoid recursion if possible.

Reply:
wasn't sure enough about dropping Self::nft_burn just yet.  since the PR is old, we can deal with this separately if needed.

Reply:
Sorry I missed this comment. Yeah, the call to `nft_burn` allows for all children & their children to be removed from the `Children` storage. It does involve recursion, but not sure what other strategy since if we remove the `Self::nft_burn` then we will be left will parent-less NFTs left in storage.